pdfly/x2pdf.py

```diff
@@ -1,12 +1,35 @@
 """Convert one or more files to PDF. Each file is a page."""
 
+import re
+
 from pathlib import Path
 from typing import List
 
 from fpdf import FPDF
 from PIL import Image
 from rich.console import Console
 
+def get_page_size(format: str):
+    """Get page dimensions based on format."""
+    sizes = {
+        "A4": (210, 297), "A3": (297, 420), "A2": (420, 594),
+        "A1": (594, 841), "A0": (841, 1189), "Letter": (215.9, 279.4),
+        "Legal": (215.9, 355.6)
+    }
+    match = re.match(r"(A\d|B\d|C\d|Letter|Legal)(-(landscape|portrait))?$", format, re.IGNORECASE)
+    if match:
+        size_key = match.group(1).upper()
+        if size_key in sizes:
+            width, height = sizes[size_key]
+            orientation = match.group(3)
+            if orientation == "landscape":
+                return (height, width)
+            elif orientation == "portrait":
+                return (width, height)
```

Comment:
```suggestion
```
Just a minor code improvement suggestion.

This change is not mandatory before merging this PR.